xarray/tests/test_backends.py

```diff
@@ -5503,9 +5502,11 @@ def create_dap2_datasets(self, **kwargs):
         # in pydap 3.5.0, urls defaults to dap2.
         url = "http://test.opendap.org/opendap/data/nc/bears.nc"
         actual = open_dataset(url, engine="pydap", **kwargs)
+        # pydap <3.5.6 converts to unicode dtype=|U. Not what
+        # xarray expects. Thus force to bytes dtype. pydap >=3.5.6
+        # does not convert to unicode. https://github.com/pydap/pydap/issues/510
+        actual["bears"].values = actual["bears"].values.astype("S")
```

Comment:
Just so I understand: pydap is changing behaviour and we are enforcing that new behaviour always?